tests/_core/test_rasterize.py

```diff
@@ -0,0 +1,44 @@
+import pytest
+
+from spatialdata._core._rasterize import rasterize
+from spatialdata._core.core_utils import get_dims, get_spatial_axes
+from tests.conftest import _get_images, _get_labels
+
+
+@pytest.mark.parametrize("_get_raster", [_get_images, _get_labels])
+def test_rasterize_raster(_get_raster):
+    rasters = _get_raster()
+    for raster in rasters.values():
+        dims = get_dims(raster)
+        spatial_dims = get_spatial_axes(dims)
+        for kwargs in [
+            {"target_unit_to_pixels": 2.0},
+            {"target_width": 10.0},
+            {"target_height": 10.0},
+            {"target_depth": 10.0},
+        ]:
+            if "z" not in dims and "target_depth" in kwargs:
+                continue
+            rasterize(
+                raster,
+                axes=spatial_dims,
+                min_coordinate=[0] * len(spatial_dims),
+                max_coordinate=[1] * len(spatial_dims),
+                target_coordinate_system="global",
+                **kwargs,
+            )
```

Comment:
It would be nice if something were done here to verify that the rasterization was done correctly.

Reply:
I'll implement this for a case without rotations since, as shown in the example, there is a problem with the precise positioning of the rasterized image, probably due to the fact that in some parts of the code the pixels coordinates refer to the (upper left) corner, while in others refer to the center of the pixels. This half pixel offset is actually large when an image is low res and covers the same area of a high res image.

Reply:
done